File: app/src/main/java/com/davenotdavid/samplepuzzle/TileImageAdapter.java
package com.davenotdavid.samplepuzzle;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import java.util.ArrayList;

public class TileImageAdapter extends BaseAdapter {
    private ArrayList<ImageView> mTileImages;
    private int mColumnWidth, mColumnHeight;

    public TileImageAdapter(ArrayList<ImageView> tileImages, int columnWidth, int columnHeight) {
        mTileImages = tileImages;
        mColumnWidth = columnWidth;
        mColumnHeight = columnHeight;
    }

    @Override
    public int getCount() {
        return mTileImages.size();
    }

    @Override
    public Object getItem(int position) {return (Object) mTileImages.get(position);}

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView tileImageView;

        if (convertView == null) {
            tileImageView = mTileImages.get(position);
        } else {
            tileImageView = (ImageView) convertView;
        }

        android.widget.AbsListView.LayoutParams params = new android.widget.AbsListView.LayoutParams(mColumnWidth, mColumnHeight);
        tileImageView.setLayoutParams(params);

        return tileImageView;
    }
}

File: app/src/main/java/com/davenotdavid/samplepuzzle/TileImageAdapter.kt
package com.davenotdavid.samplepuzzle

import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.BaseAdapter
import android.widget.ImageView

class TileImageAdapter(private val tileImages: List<ImageView>,
                       private val boardColumnWidth: Int,
                       private val boardColumnHeight: Int) : BaseAdapter() {

    override fun getCount(): Int = tileImages.size

    override fun getItem(position: Int): ImageView = tileImages[position]

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val tileImageView = if (convertView == null) {
            tileImages[position]
        } else {
            convertView as ImageView
        }
        tileImageView.layoutParams = AbsListView.LayoutParams(boardColumnWidth, boardColumnHeight)

        return tileImageView
    }
}
